activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -944,6 +944,8 @@ def where!(opts, *rest) # :nodoc:
     # This is short-hand for <tt>unscope(where: conditions.keys).where(conditions)</tt>.
     # Note that unlike reorder, we're only unscoping the named conditions -- not the entire where statement.
     def rewhere(conditions)
+      return spawn.unscope(:where) if conditions.nil?
```

Comment:
`unscope` calls `spawn` already.

```suggestion
      return unscope(:where) if conditions.nil?
```

Reply:
@byroot 
thank you for your review 🙇‍♀️ 
I fixed it.